osquery/tables/system/linux/cpu_time.cpp

```diff
@@ -0,0 +1,86 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+#include <vector>
+#include <boost/filesystem.hpp>
+#include <boost/algorithm/string/trim.hpp>
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/algorithm/string/split.hpp>
+
+#include <osquery/core.h>
+#include <osquery/tables.h>
+#include <osquery/filesystem.h>
+
+#include <osquery/core/conversions.h>
+
+namespace osquery {
+namespace tables {
+
+const std::string kProcStat = "/proc/stat";
+
+std::vector<std::string> procFromFile(const std::string path) {
+    std::string content;
+    std::vector<std::string> proc_lines;
+
+    if (!isReadable(path).ok()) {
+        return proc_lines;
+    }
+
+    if (!forensicReadFile(path, content).ok()) {
+        return proc_lines;
+    }
+
+    auto lines = split(content, "\n");
+
+    for (auto& line : lines) {
+        boost::trim(line);
+        if (boost::starts_with(line, "cpu")) {
+            proc_lines.push_back(line);
+        }
+    }
+
+    // Remove first cpu line which doesn't give specific core information
+    proc_lines.erase(proc_lines.begin());
+
+    return proc_lines;
+}
+
+Row genCpuTimeLine(std::string line) {
+    Row r;
+
+    std::vector<std::string> words;
+    split(words, line, boost::is_any_of(" "));
+
+    r["core"] = words[0].back();
+    r["user"] = words[1];
+    r["nice"] = words[2];
+    r["system"] = words[3];
+    r["idle"] = words[4];
+    r["iowait"] = words[5];
+    r["irq"] = words[6];
+    r["softirq"] = words[7];
+    r["steal"] = words[8];
+    r["guest"] = words[9];
+    r["guest_nice"] = words[10];
```

Comment:
What if there are only 10 values after the split? As this stands there wont be a problem, since the `operator[]` will create a `&&` value for you. Unlike the `.back()` access above, this wont except. However, it might be nice to check the size of the `words` immediately following the `osquery::split` call.

Something like:
```cpp
if (words.size() < 11) {
  // We're going to have a problem.
  return {};
}
```